main.go

```diff
@@ -11,30 +15,87 @@ const (
 
 // generateRandomElements generates random elements.
 func generateRandomElements(size int) []int {
-	// ваш код здесь
+	if size <= 0 {
+		return []int{}
+	}
+
+	data := make([]int, size)
+	rand.New(rand.NewSource(time.Now().UnixNano()))
+
+	for i := range size {
+		data[i] = rand.Intn(size * 10)
+	}
+	return data
 }
 
 // maximum returns the maximum number of elements.
 func maximum(data []int) int {
-	// ваш код здесь
+	if len(data) == 0 {
+		return 0
+	}
+
+	max := data[0]
+	for _, num := range data {
+		if num > max {
+			max = num
+		}
+	}
+	return max
 }
 
 // maxChunks returns the maximum number of elements in a chunks.
+
 func maxChunks(data []int) int {
-	// ваш код здесь
+	if len(data) == 0 {
+		return 0
+	}
+
+	results := make([]int, CHUNKS)
+	var wg sync.WaitGroup
+	wg.Add(CHUNKS)
+
+	for i := range CHUNKS {
+		go func(idx int) {
+			defer wg.Done()
+
+			start := idx * len(data) / CHUNKS
+			end := (idx + 1) * len(data) / CHUNKS
+
+			if start >= end {
+				results[idx] = math.MinInt
+				return
+			}
+
+			chunk := data[start:end]
+			chunkMax := chunk[0]
```

Comment:
У тебя уже есть расчеты максимум
```suggestion
			chunkMax := maximum(chunk)
```